internal/js_parser/ts_parser_test.go

```diff
@@ -158,6 +158,29 @@ func TestTSTypes(t *testing.T) {
 	expectPrintedTS(t, "type Foo<T> = {readonly [P in keyof T]: T[P]}", "")
 	expectPrintedTS(t, "type Foo<T> = {-readonly [P in keyof T]: T[P]}", "")
 	expectPrintedTS(t, "type Foo<T> = {+readonly [P in keyof T]: T[P]}", "")
+	expectPrintedTS(t, "type Foo<out T> = T;", "")
+	expectPrintedTS(t, "type Foo<in T> = keyof T;", "")
+	expectPrintedTS(t, "type Foo<out T, out K extends keyof T> = T[K];", "")
+	expectPrintedTS(t, "type Foo<in out T> = T[keyof T];", "")
+	expectPrintedTS(t, "type Foo<in T> = keyof T;", "")
+	expectPrintedTS(t, "interface Baz<out T> {}", "")
+	expectPrintedTS(t, "interface Baz<in T> {}", "")
+	expectPrintedTSX(t, "<in T></in>", "/* @__PURE__ */ React.createElement(\"in\", {\n  T: true\n});\n")
+	expectPrintedTSX(t, "<out T></out>", "/* @__PURE__ */ React.createElement(\"out\", {\n  T: true\n});\n")
+	expectPrintedTSX(t, "<in out T></in>", "/* @__PURE__ */ React.createElement(\"in\", {\n  out: true,\n  T: true\n});\n")
+	expectPrintedTSX(t, "<in T extends={true}></in>", "/* @__PURE__ */ React.createElement(\"in\", {\n  T: true,\n  extends: true\n});\n")
+	expectPrintedTSX(t, "<out T extends={true}></out>", "/* @__PURE__ */ React.createElement(\"out\", {\n  T: true,\n  extends: true\n});\n")
+	expectPrintedTSX(t, "<in out T extends={true}></in>", "/* @__PURE__ */ React.createElement(\"in\", {\n  out: true,\n  T: true,\n  extends: true\n});\n")
+	expectPrintedTSX(t, "<in T,>() => {}", "() => {\n};\n")
```

Comment:
Where did these test cases come from? I can't reproduce this. The [latest nightly build of the TypeScript compiler](https://www.typescriptlang.org/play?ts=4.7.0-dev.20220331#code/DwSwdgBAKgNAfACgJQQLxwgbwL5A) gives this for `<in T,>() => {}` instead when JSX is enabled:

```js
React.createElement("in", { T: true }, "() => ");
```

and this when JSX is disabled:

```js
T,  > ();
{ }
```

Reply:
Oh, thanks.
I will report this issue to TypeScript repo.

Reply:
You are right.
It's TypeScript Error `TS1273`.

I have modified the code and added some more test cases.